src/core/scope.ts

```diff
@@ -39,7 +41,9 @@ function createScope(
 		// Whenever the modified draft contains a draft from another scope, we
 		// need to prevent auto-freezing so the unowned draft can be finalized.
 		canAutoFreeze_: true,
-		unfinalizedDrafts_: 0
+		unfinalizedDrafts_: 0,
+		handledSet_: new WeakSet(),
```

Comment:
since these scopes are thrown away at the end of the process, normal Sets should work and might maybe more efficient?